171784102 Add Create Accommodation facility API Resource

Francois-MUGOROZI opened the pull request:
### What the PR does?
As travel admin or supplier should be able to create accommodation in Barefoot Nomad
### Description of the task to be completed
- add create accommodation endpoint so that permitted user can add new accommodation record
- add get accommodation endpoint so that permitted user can view all accommodations in Barefoot Nomad
- add get single accommodation so that user can view accommodation details by providing its id
### How this should be tested?
Having your environment setup (Postgres DB, node), clone the repo and cd into it,
run "npm install" then run "npm run start:dev". Open postman 

1. To create accommodation: make post request at /api/v1/accommodations with required data.
2.  To get all accommodation: make get request at /api/v1/accommodations
3. To get single accommodation: make get request at /api/v1/accommodations/{id}
4. To get all accommodations in given city as Requester: make get request at /ap/v1/accommodations?city={city}
With browser navigate to /api-docs for API documentation
Make sure you have set authorization headers with an access token
### Relevant Pivotal Tracker story
[171784102](https://www.pivotaltracker.com/n/projects/2438874/stories/171784102)

### Screenshots
![Capture](https://user-images.githubusercontent.com/53614123/80307063-57aa7880-8773-11ea-9612-dd24955dfff2.JPG)
![Capture2](https://user-images.githubusercontent.com/53614123/80307075-685aee80-8773-11ea-986c-ff914f923856.JPG)

kabundege reviewed (commented):
Squash your commit messages, solve the conflicts and If possible you could share with us some response images.

Francois-MUGOROZI committed:
ft(accommodation): add create accommodation facility
- added create accomm endpoint
- added view accomm endpoint
- added tests and docs
[Finish #171784102]

irfiacre commented on server/controllers/accomodationController.js:
```diff
@@ -0,0 +1,107 @@
+      'accommodation',
+    );
+
+    const message = 'Accommodation crated successfully';
```
I think it could be better to user `created` instead of `crated`.

irfiacre commented on server/helpers/accommodationFilter.js:
```diff
@@ -0,0 +1,17 @@
+const filtereAccommodation = (data) => {
```
Similar to here maybe `filterAccomodation` may work better.

irfiacre commented on server/models/accommodations.js:
```diff
@@ -0,0 +1,59 @@
+  accommodations.associate = (models) => {
+    accommodations.hasMany(models.photos, {
+      foreignKey: 'ownerId',
+      as: 'accomPhotos',
```
We could use `accommodationPhotos` this would be okay.

irfiacre commented on server/routes/api/accommodations/accommRoute.js:
```diff
@@ -0,0 +1,45 @@
+);
+
+router.get(
+  '/city/accommodations/:city',
```
Here I think the overall route would be `/api/v1/accommodations/city/accomodations/:city` what if we could use `/api/v1/accommodations/city/:city`

bdushimi replied:
I would suggest to go with /api/v1/accommodations?city="cityName"

irfiacre commented on server/routes/api/accommodations/cityAccommRoute.js:
```diff
@@ -0,0 +1,19 @@
+const router = express.Router();
+
+router.get(
+    '/city/accommodations/:city',
```
The same here.

bdushimi replied:
Kindly have all routes related to accommodation in one file.

irfiacre reviewed (commented):
Great job on the work you've done @Francois-MUGOROZI; just some minor suggestions that I think would make your feature better.

Francois-MUGOROZI committed:
ft(accommodation): add create accommodation facility
    - added create accomm endpoint
    - added view accomm endpoint
    - added tests and docs
    [Finish #171784102]
